lib.php

```diff
@@ -600,6 +603,53 @@ private function compress($value) {
         }
     }
 
+    /**
+     * Create a new RedisCluster instance already connect to the server.
+     *
+     * @param $servers string The server connection strings separated by newlines.
+     * @return RedisCluster
+     */
+    private function new_redis_cluster($servers) {
+        $servers = explode("\n", $servers);
+
+        $trimmedservers = [];
+        foreach ($servers as $server) {
+            $server = trim($server);
+            if (!empty($server)) {
+                $trimmedservers[] = $server;
+            }
+        }
+
+        $redis = new RedisCluster(null, $trimmedservers);
+
+        $this->isready = true; // RedisCluster will throw an exception if cannot connect.
```

Comment:
There is no way to go around it: if you Moodle cannot reach the redis server it will die.

If not using cluster, when it cannot connect it returns false at first, but afterwards it will throw an exception 'Redis server went away' as soon as you try to interact with the redis.

In case of using cluster, because we get a nice exception (reason why it failed), I decided to let it get thrown as it is more informative than the later 'Redis server went away' if we catch it.